src/frontends/pytorch/src/utils.cpp

```diff
@@ -591,6 +592,13 @@ Output<Node> masked_fill(ov::pass::NodeRegistry& rg,
     return rg.make<opset10::Select>(bool_mask, _value, data);
 }
 
+Output<Node> masked_select(ov::pass::NodeRegistry& rg,
+                         const Output<Node>& data,
+                         const Output<Node>& mask) {
+    auto _index = rg.make<opset10::NonZero>(mask);
+    return rg.make<opset10::GatherND>(data, _index);
```

Comment:
You need to do `Transpose` after `NonZero`, since the format of indices in `GatherND` is different. Like here:
https://github.com/openvinotoolkit/openvino/blob/c20d2327316162667ea1f92f947a2bed6303fc4e/src/frontends/pytorch/src/op/index.cpp#L241